src/daemon/metric.h

```diff
@@ -70,6 +70,19 @@ typedef struct {
   size_t num;
 } label_set_t;
 
+/* label_set_clone copies all the laels in src into dest. If dest contains
+ * any labels prior to calling label_set_clone, the associated memory is
+ * leaked. */
+int label_set_clone(label_set_t *dest, label_set_t src);
```

Comment:
That seems rather nasty way of introducing accidental leaks.

What if it would return e.g. `EEXISTS` if there are already labels in it, like `label_set_add()` does?

(Maybe also log it, or assert on it?)

Reply:
Changed the semantic to return `EINVAL` if `dest` is not empty.